include/utap/typechecker.h

```diff
@@ -69,7 +69,7 @@ class TypeChecker : public DocumentVisitor, public AbstractStatementVisitor
 
     expression_t checkInitialiser(type_t type, expression_t init);
     bool areAssignmentCompatible(type_t lvalue, type_t rvalue, bool init = false) const;
-    bool areInlineIfCompatible(type_t thenArg, type_t elseArg) const;
+    bool areInlineIfCompatible(type_t result_type, type_t thenArg, type_t elseArg) const;
```

Comment:
The ternary operator **converts** the branching types into the common type, then the common type gets converted into the result type -- it's a very verbose procedure and I think we should stick to it:
https://en.cppreference.com/w/c/language/operator_other#Conditional_operator 

So a better procedure is to compute a common type and then use `areAssignmentCompatible` to check both branches.